libhb/encavcodec.c

```diff
@@ -1164,6 +1265,50 @@ int encavcodecWork( hb_work_object_t * w, hb_buffer_t ** buf_in,
         return HB_WORK_DONE;
     }
 
+#if HB_PROJECT_FEATURE_QSV
+    // postponed encoder initialization, reused code from encavcodecInit()
+    if (hb_hwaccel_is_full_hardware_pipeline_enabled(pv->job) &&
```

Comment:
not the prettiest piece of code, but the initialization logic in ffmpeg and handbrake is different. in case of two separate contexts, it requires copying frames each time, which is quite expensive.